### ISSUE TITLE
Extensions are not applied

### ISSUE THREAD

**Issue by ddorstijn** [riccardoperra/solid-codemirror#12]
Hello,

Thanks for the solid bindings. I wanted to try out the library, so I took the extensions example from the readme and pasted it in a fresh project. However, the result is a minimal code editor without any of the extensions applied. Could there be a bug?

The code:
``` typescript
import { createCodeMirror } from 'solid-codemirror';
import { createSignal } from 'solid-js';
import { EditorView, lineNumbers } from '@codemirror/view';

export default function App() {
  const [code, setCode] = createSignal("console.log('hello world!')");
  const [showLineNumber, setShowLineNumber] = createSignal(true);
  const { ref, createExtension } = createCodeMirror({ onValueChange: setCode });

  const theme = EditorView.theme({
    '&': {
      background: 'red'
    }
  });

  // Add a static custom theme
  createExtension(theme);

  // Toggle extension
  createExtension(() => showLineNumber() ? lineNumbers() : []);

  // Remove line numbers after 2.5s
  setTimeout(() => {
    setShowLineNumber(false);
  }, 2500);

  return <div ref={ref} />
}
```

Output:
![afbeelding](https://user-images.githubusercontent.com/1730282/197536479-ba118fb5-6508-4b73-b1a2-480becbe0590.png)

**Comment by riccardoperra** [comment]
Hey @ddorstijn, thanks for reaching me 😄 

I'm a bit confused of why this is not working..I find a solution of this but I'll will dive into it to fix this issue

https://stackblitz.com/edit/solidjs-templates-eosgqx?file=src%2FApp.tsx

You can check this playground I did with the solution 😄 

All consist of mounting the editor ref inside `onMount` instead of using directly the solid "ref" accessor

**Comment by ddorstijn** [comment]
Hey @riccardoperra,

Thanks for the quick response. I just figured it out as well. You need to set the ref by onMount(). This was missing from the readme. I figured it out by looking at the codeimage implementation.

**Comment by riccardoperra** [comment]
Hey @ddorstijn, I have released the fix with [2.1.0 tag version](https://github.com/riccardoperra/solid-codemirror/releases/tag/solid-codemirror%402.1.0) 😄.

It concerned some differences on the effects execution with the latest version of solidjs.

For this reason I have added solid-js@1.6.0 as a minimum version in peerDependencies
